plugins/foreman_discovery/2.0/index.md

```diff
@@ -0,0 +1,306 @@
+
+Alternatively, the Discovery image also searches for DNS SRV record named
+_x-foreman._tcp. If the DNS server is configured for this (see the example for
+ISC BIND below), then it is not required to provide *foreman.url* at all, but
+*proxy.type* is still required.
+
+Here is an example of ISC BIND configuration statement to set *foreman* A
+record to be used with HTTPS protocol.
+
+    _x-foreman._tcp SRV 0 5 443 foreman
+
+You can mix both approaches and override SRV record with the command line.
+
+It is important to keep *IPAPPEND 2* option which is key configuration option
+to detect interface connected to provisioning network. Also never change or
+remove *root* option, otherwise image will not boot properly.
+
+It is important to know that DNS servers from DHCP are taken into acount for
+only for the interface that was specified via the BOOTIF option. This means
+when a system have multiple NICs, DNS will work for the correct interface - the
+one that was booted from.
+
+## 3.1.2 Update PXE default template
+
+Once the template has been updated, click the *Build PXE Default* button at the
+top of the Provisioning Templates page. This will instruct the TFTP proxy to
+rewrite the *pxelinux.cfg/default* file.
+
+Repeat this step every time a change is done on the default template.
+
+## 3.1.3 Global settings
+
+If Locations and/or Organisations are enabled, Foreman will default to using
+the first Location and first Organisation for Discovered hosts. If hosts should
+be placed in some other Location/Organization, alter the default settings in
+*More->Settings->Discovered*.
+
+There is also setting called *discovery_fact* which defaults to
+*discovery_bootif*. It specify which incoming fact should be used to get
+the MAC address. By default PXELinux BOOTIF kernel command line option is used
+which gives a MAC address of the interface which was booted from.
+
+Note that *auto_discovery* option is set to *false* by default. If you want to
+trigger provisioning automatically with rules, you need to turn this setting
+on. It's recommended to try provision manually first to test before proceeding.
+
+## 3.2 Foreman Proxy Discovery plugin
+
+Make sure *foreman_url* setting is present in the Foreman Proxy configuration file.
+
+    # grep foreman_url /etc/foreman-proxy/settings.yml 
+    :foreman_url: https://FOREMAN_HOST
+
+This should by done automatically by our installer. It is a good idea to check
+if the host responds properly and there are no firewalls blocking the
+communication.
+
+# 4. Usage
+
+Boot a machine in any provisioning network that was configured with the default
+PXE configuration above. It should register with Foreman and appear in *Hosts
+-> Discovered Hosts*. Select it and choose Provision. This will redirect to the
+normal Edit page for a Host, with the discovered data filled in where possible.
+Fill in the details as normal.
+
+On save, a reboot request is sent to the discovered host, after which it should
```

Comment:
It may be useful to mention that provisioning the host will typically write a PXELinux menu that makes the build process work?